core/google/api/core/retry.py

```diff
@@ -199,6 +200,34 @@ def retry_target(target, predicate, sleep_generator, deadline, on_error=None):
     raise ValueError('Sleep generator stopped yielding sleep values.')
 
 
+class RetryOptions(
```

Comment:
We intentionally do not use a separate retry options class, instead, preferring to just allow the user to pass in a `Retry` instance.

See the discussion here: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/4148#issuecomment-335523296

Reply:
Thanks. I created the `RetryOptions` because to the user of `mutate_rows`, requiring the `Retry` instance would be impractical since they don't know what the `predicate` should be. The API method doesn't raise but return an error code for a corresponding row in case of an error. For them, only the retry parameters are meaningful.

Having said that, I understand the discussion. Will revert the `retry.py` in the next commit.

Reply:
See the lower conversation in that same PR - BigQuery had a similar need to have its own predicate. I would suggest making the default retry a global constant so users can do:

```
retry = bigtable.DEFAULT_RETRY.with_deadline(60)
```

Reply:
Yes. I plan to do a `MUTATE_ROWS_DEFAULT_RETRY` which the user can customize with a desired `deadline`. Other retry params are more like impl detail for retrying, so less important to the user I guess.